typeid/typeid-go/README.md

```diff
@@ -1,10 +1,63 @@
 # TypeID Go
 ### A golang implementation of [TypeIDs](https://github.com/jetpack-io/typeid)
-![License: Apache 2.0](https://img.shields.io/github/license/jetpack-io/typeid-go)
+![License: Apache 2.0](https://img.shields.io/github/license/jetpack-io/typeid-go) [![Go Reference](https://pkg.go.dev/badge/go.jetpack.io/typeid.svg)](https://pkg.go.dev/go.jetpack.io/typeid)
 
 TypeIDs are a modern, **type-safe**, globally unique identifier based on the upcoming
 UUIDv7 standard. They provide a ton of nice properties that make them a great choice
 as the primary identifiers for your data in a database, APIs, and distributed systems.
 Read more about TypeIDs in their [spec](https://github.com/jetpack-io/typeid).
 
-This particular implementation provides a go library for generating and parsing TypeIDs.
\ No newline at end of file
+This particular implementation provides a go library for generating and parsing TypeIDs.
+
+## Installation
+
+To add this library as a dependency in your go module, run:
+
+```bash
+go get go.jetpack.io/typeid
+```
+
+## Usage
+This library provides both a statically typed and a dynamically typed version of TypeIDs.
+
+The statically typed version that lives under the `typed` package is the recommended version,
+because it makes it possible for the go compiler itself to enforce type safety.
+
+To use it, first define your TypeID types:
+
+```go
+import (
+  typeid "go.jetpack.io/typeid/typed"
+)
+
+type UserIDType struct{}
+func (UserIDType) Type() string { return "user" }
```

Comment:
This is cool, but should we recommend it? It makes stuff more verbose which makes the library a bit harder to adopt.